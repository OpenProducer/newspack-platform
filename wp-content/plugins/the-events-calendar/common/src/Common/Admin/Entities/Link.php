<?php
/**
 * Link element.
 *
 * @since 6.1.0
 */

declare( strict_types=1 );

namespace TEC\Common\Admin\Entities;

use Tribe\Utils\Element_Attributes as Attributes;
use Tribe\Utils\Element_Classes as Classes;

/**
 * Class Link
 *
 * @since 6.1.0
 */
class Link extends Base_Entity {

	/**
	 * The URL for the link.
	 *
	 * @var string
	 */
	private string $url = '';

	/**
	 * The text for the link.
	 *
	 * @var string
	 */
	private string $text = '';

	/**
	 * Link constructor.
	 *
	 * @since 6.1.0
	 *
	 * @param string      $url        The URL for the link.
	 * @param string      $text       The text for the link.
	 * @param ?Classes    $classes    The classes for the link.
	 * @param ?Attributes $attributes The attributes for the link.
	 */
	public function __construct( string $url, string $text, ?Classes $classes = null, ?Attributes $attributes = null ) {
		$this->url  = $url;
		$this->text = $text;

		if ( $classes ) {
			$this->set_classes( $classes );
		}

		if ( $attributes ) {
			$this->set_attributes( $attributes );
		}
	}

	/**
	 * Render the link.
	 *
	 * @since 6.1.0
	 *
	 * @return void
	 */
	public function render() {
		printf(
			'<a href="%s" class="%s" %s>%s</a>',
			esc_url( $this->url ),
			esc_attr( $this->get_classes() ),
			$this->get_attributes(), // phpcs:ignore StellarWP.XSS.EscapeOutput,WordPress.Security.EscapeOutput
			esc_html( $this->text )
		);
	}
}
