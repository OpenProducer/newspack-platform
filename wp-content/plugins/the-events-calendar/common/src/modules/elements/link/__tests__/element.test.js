/**
 * External dependencies
 */
import React from 'react';

/**
 * Internal dependencies
 */
import Link from '../element.js';

describe( 'Link Element', () => {
	it( 'renders link', () => {
		const component = renderer.create(
			<Link href="#">label</Link>
		);
		expect( component.toJSON() ).toMatchSnapshot();
	} );

	it( 'renders button with class', () => {
		const component = renderer.create(
			<Link className="test-class" href="#">label</Link>
		);
		expect( component.toJSON() ).toMatchSnapshot();
	} );

	it( 'renders button with target', () => {
		const component = renderer.create(
			<Link href="#" target="_blank">label</Link>
		);
		expect( component.toJSON() ).toMatchSnapshot();
	} );

	it( 'renders button with prop', () => {
		const component = renderer.create(
			<Link href="#" title="title">label</Link>
		);
		expect( component.toJSON() ).toMatchSnapshot();
	} );
} );
