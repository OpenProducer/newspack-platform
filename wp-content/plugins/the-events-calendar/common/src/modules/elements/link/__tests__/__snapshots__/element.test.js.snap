// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Link Element renders button with class 1`] = `
<a
  className="tribe-editor__link test-class"
  href="#"
>
  label
</a>
`;

exports[`Link Element renders button with prop 1`] = `
<a
  className="tribe-editor__link"
  href="#"
  title="title"
>
  label
</a>
`;

exports[`Link Element renders button with target 1`] = `
<a
  className="tribe-editor__link"
  href="#"
  rel="noopener noreferrer"
  target="_blank"
>
  label
</a>
`;

exports[`Link Element renders link 1`] = `
<a
  className="tribe-editor__link"
  href="#"
>
  label
</a>
`;
